app/Console/Commands/ImportOpesCommand.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Console\Commands;
+
+class ImportOpesCommand extends ImportCommand
+{
+    /**
+     * The name of the console command.
+     *
+     * @var string
+     */
+    protected $name = 'import:opes';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Import data for "Opes"';
+
+    /**
+     * Process a single value. This method can be overriden in an importer
+     * to do some kind of processing before the data are inserted into the database,
+     * e.g. converting a string value to a JSON value.
+     *
+     * @return array
+     */
+    protected function processValue(string $column, string $value)
+    {
+        $value = trim($value, " \t\n\r\0\x0B\"");
+
+        switch ($column) {
+            case 'persons':
+            case 'subj_headings':
+                $value = explode(';', $value);
+                $value = array_map(function($x) { return trim($x); }, $value);
+                $value = json_encode($value);
+                break;
+
+            case 'negative_in_copenhagen':
+                $value = (substr($value, 0, 3) === 'Yes') ? '1' : '0';
+                break;
+
+            case 'inv_no':
+                // TODO
+                // Strip "P.Oslo inv." ???
+                break;
+
+            case 'date1':
+            case 'date2':
+                if ($value !== '') {
+                    //$value = preg_replace('/\/.*$/', '', $value);
+                    //$value = preg_replace('/\(\?\)/', '', $value);
+                    if (!is_numeric($value)) {
+                        echo "'$column' is not numeric: $value\n";
+                    }
+                }
+                break;
+
+            case 'date_cataloged':
+                if ($value !== '') {
+                    $value = explode('.', $value);
+                    if (count($value) != 3) {
+                        print("INVALID DATE: " . implode('.', $value));
+                        die;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method processValue() contains an exit expression.](https://app.codacy.com/manual/scriptotek/ub-baser/pullRequest?prid=4835963)